### DISCUSSION TITLE
Support TLD

### DISCUSSION THREAD

**Discussion by oxilor** [vercel/next.js#22682] [Ideas]
**Describe the feature you'd like to request**

A way to configure i18n so the locale will be detected by the top level domain name (com - en, ru - ru). This is necessary when the app is shared by multiple services located in different domains. 

For example, we have a single instance of the blog app for multiple services:
* `app1.com/blog` – The blog app for the `app1` service in English.
* `app1.ru/blog` – The blog app for the `app1` service in Russian.
* `app2.com/blog` – The blog app for the `app2` service in English.
* `app2.ru/blog` – The blog app for the `app2` service in Russian.

**Describe the solution you'd like**

The `next.config.js` file may look like this:
```js
module.exports = {
  i18n: {
    locales: ['en', 'ru'],
    defaultLocale: 'en',
    localeDetection: false,
    domains: [
      {
        domain: 'com',
        defaultLocale: 'en',
      },
      {
        domain: 'ru',
        defaultLocale: 'ru',
      },
    ],
  },
};
```

**Describe alternatives you've considered**

-

**Comment by oxilor** [comment]
It may be better to support a custom function in the `domain` field. For example:
```js
module.exports = {
  i18n: {
    locales: ['en', 'ru'],
    defaultLocale: 'en',
    localeDetection: false,
    domains: [
      {
        domain: (hostname) => hostname.endsWith('.com'),
        defaultLocale: 'en',
      },
      {
        domain: (hostname) => hostname.endsWith('.ru'),
        defaultLocale: 'ru',
      },
    ],
  },
};
```
Thus, it will be possible to set any logic to determine whether a hostname should use a specific default locale.

**Comment by z4nr34l** [comment]
I think this need to work as described in [i18n Next.js docs](https://nextjs.org/docs/advanced-features/i18n-routing) as providing only TLD can be potential phishing possibility in case of small mistake, as well as can make some problems when you want to script your CORS settings. Many next.js websites have more than 1 environment which have different URLs, and i18n based on TLD is not good thing on dev/stagging environments.
